src/components/buttons/default.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+import colors from '../../assets/colors'
+import * as fonts from '../../assets/fonts'
+
+interface ButtonProps extends React.HTMLProps<Button> {
+    state?: ButtonState
+    block?: boolean
+}
+
+export class ButtonState extends String {
+    static Default: ButtonState = "default"
+    static Loading: ButtonState = "loading"
+    static Success: ButtonState = "success"
+    static Failure: ButtonState = "failure"
+}
+
+class Button extends React.Component<ButtonProps, any> {
+    public static defaultProps: ButtonProps
+
+    render(): JSX.Element {
+        const newProps: any = {...this.props}
```

Comment:
Yes, that syntax is supported and I agree it's a bit more readable but I needed to "cast" props to type `any` to silence the compiler when we pass props down to `<button>` or `<a>`